test/UsageDemo.cpp

```diff
@@ -0,0 +1,113 @@
+#include <memory>
+#include <array>
+#include <vector>
+
+#include "Reader.hpp"
+
+#include <boost/test/unit_test.hpp>
+
+
+// These classes are not intended for production use. The are merely intended
+// to illustrate where data is stored in the "block" data structures.
+class Bond
+{
+public:
+    Bond (int atom0, int atom1, int bond_order): atom0(atom0), atom1(atom1), bond_order(bond_order) {}
+
+    const int atom0;
+    const int atom1;
+    const int bond_order;
+};
+
+class Structure
+{
+public:
+    std::string title;
+    std::vector<int> atomic_numbers;
+    std::vector<std::array<double, 3>> coordinates;
+    std::vector<Bond> bonds;
+};
+
+
+BOOST_AUTO_TEST_SUITE(DemoSuite)
+
+// Reads all atom and bond information from test.mae, which is a standard
+// mae formatted file. Only accesses properties that are gauranteed to
+// exist in every f_m_ct block.
+BOOST_AUTO_TEST_CASE(maeBlock)
+{
+    std::ifstream ss("test.mae");
+    schrodinger::mae::Reader r(ss);
+
+    std::vector<std::shared_ptr<Structure>> structures;
+    std::shared_ptr<schrodinger::mae::Block> b;
+    while ((b = r.next("f_m_ct")) != nullptr) {
+        auto st = std::make_shared<Structure>();
+        st->title = b->getStringProperty("s_m_title");
+
+        // Atom data is in the m_atom indexed block
+        {
+            const auto atom_data = b->getIndexedBlock("m_atom");
+            // All atoms are gauranteed to have these three field names:
+            const auto atomic_numbers = atom_data->getIntProperty("i_m_atomic_number");
+            const auto xs = atom_data->getRealProperty("r_m_x_coord");
+            const auto ys = atom_data->getRealProperty("r_m_y_coord");
+            const auto zs = atom_data->getRealProperty("r_m_z_coord");
+            const auto size = atomic_numbers->size();
+            BOOST_REQUIRE_EQUAL(size, xs->size());
+            BOOST_REQUIRE_EQUAL(size, ys->size());
+            BOOST_REQUIRE_EQUAL(size, zs->size());
+
+            // atomic numbers, and x, y, and z coordinates
+            for (size_t i=0; i<size; ++i) {
+                st->atomic_numbers.push_back(atomic_numbers->at(i));
+                st->coordinates.push_back({{xs->at(i), ys->at(i), zs->at(i)}});
+
+                // Other properties could fail, because not all properties have
+                // values for all atoms.
+            }
+        }
+        BOOST_CHECK_GT(st->atomic_numbers.size(), 0u);
+
+
+        // Bond data is in the m_bond indexed block
+        {
+            const auto bond_data = b->getIndexedBlock("m_bond");
+            // All bonds are gauranteed to have these three field names:
+            auto from_atoms = bond_data->getIntProperty("i_m_from");
+            auto to_atoms = bond_data->getIntProperty("i_m_to");
+            auto orders = bond_data->getIntProperty("i_m_order");
+            const auto size = from_atoms->size();
+
+            for (size_t i=0; i<size; ++i) {
+                // Maestro atoms are 1 indexed!
```

Comment:
I think this would be more clear if you said "Atom indices in the bond data are 1-based!"